homeassistant/components/sensor/nordpool.py

```diff
@@ -0,0 +1,252 @@
+# -*- coding: utf-8 -*-
+"""
+Sensor to display the current power prices from NordPool.
+
+To enable the sensor, add this to the config.yaml file under sensors:
+
+sensor:
+  - platform: nordpool
+    currency: 'NOK'
+    region: 'Kr.sand'
+
+The available currencies are EUR, DKK, NOK and SEK.
+
+The available regions are DK1, DK2, EE, FI, LT, LV, Oslo, Kr.sand, Bergen,
+Molde, Tr.heim, Tromsø, SE1, SE2, SE3, SE4 and SYS.
+
+Optional parameters are:
+    name: 'Elspot kWh'
+    offset: '01:00:00'
+
+You may use the offset to adjust the "current price" to indicate the price
+at an offset. Positive offset give you future prices, and negative offset
+give you past prices.
+"""
+import datetime
+import logging
+# pylint: disable=E0611
+from random import randrange
+# pylint: enable=E0611
+import requests
+import voluptuous
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (ATTR_ATTRIBUTION, CONF_CURRENCY, CONF_OFFSET,
+                                 CONF_REGION, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+_CURRENCY_LIST = ['DKK', 'EUR', 'NOK', 'SEK']
+
+_CURRENCY_FRACTION = {
+    'DKK': 'Øre',
+    'EUR': 'Cent',
+    'NOK': 'Øre',
+    'SEK': 'Öre'
+}
+
+_REGION_NAME = ['DK1', 'DK2', 'EE', 'FI', 'LT', 'LV', 'Oslo', 'Kr.sand',
+                'Bergen', 'Molde', 'Tr.heim', 'Tromsø', 'SE1', 'SE2', 'SE3',
+                'SE4', 'SYS']
+
+DEFAULT_CURRENCY = 'EUR'
+DEFAULT_REGION = 'SYS'
+DEFAULT_NAME = 'Elspot kWh'
+
+_TODAY = 0
+_TOMORROW = 1
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    voluptuous.Optional(CONF_CURRENCY, default=DEFAULT_CURRENCY):
+        voluptuous.In(_CURRENCY_LIST),
+    voluptuous.Optional(CONF_REGION, default=DEFAULT_REGION):
+        voluptuous.In(_REGION_NAME),
+    voluptuous.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    voluptuous.Optional(CONF_OFFSET, default=0): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Get the platform settings and setup the sensor."""
+    currency = config.get(CONF_CURRENCY)
+    region = config.get(CONF_REGION)
+    name = config.get(CONF_NAME)
+    offset = config.get(CONF_OFFSET)
+    add_devices([Nordpool(name, currency, region, offset)])
+
+
+class Nordpool(Entity):
+    """The Nordpool platform."""
+
+    def __init__(self, name, currency, region, offset):
+        """Initialize the sensor."""
+        self._prices = [None, None, None, None, None, None, None, None,
+                        None, None, None, None, None, None, None, None,
+                        None, None, None, None, None, None, None, None,
+                        0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
```

Comment:
Why first None and so 0.0

Reply:
The old version did not have the available property, so I had to find a way to prevent array out of bounds error. I'll implement the nordpool library from pypi and rewrite the code to do different tests on valid data.